plugin/speedb/memtable/hash_spd_rep.cc

```diff
@@ -703,19 +703,36 @@ class HashLocklessRep : public MemTableRep {
 
   std::shared_ptr<SortVectorContainer> sort_vectors_cont_;
 
-  size_t GetHash(const char* key) const {
-    Slice slice = UserKey(key);
-    return MurmurHash(slice.data(), static_cast<int>(slice.size()), 0) %
-           bucket_size_;
+  static size_t GetHash(const Slice& user_key_without_ts) {
+    return MurmurHash(user_key_without_ts.data(),
+                      static_cast<int>(user_key_without_ts.size()), 0);
+  }
+
+  static Slice UserKeyWithoutTimestamp(
+      const Slice internal_key, const MemTableRep::KeyComparator& compare) {
+    auto key_comparator = static_cast<const MemTable::KeyComparator*>(&compare);
```

Comment:
Can we use `dynamic_cast` and assert here if `ROCKSDB_USE_RTTI` is set and `NDEBUG` isn't, just to be on the safe side?

```suggestion
#if defined(ROCKSDB_USE_RTTI) && !defined(NDEBUG)
    auto key_comparator = dynamic_cast<const MemTable::KeyComparator*>(&compare);
    assert(key_comparator != nullptr);
#else
    auto key_comparator = static_cast<const MemTable::KeyComparator*>(&compare);
#endif
```

Reply:
done